#! /bin/sh -e
## 50_check_for_charsetp by Jens Peter Secher <jps@debian.org>

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0
--- sml-mode-4.0.orig/sml-mode.el
+++ sml-mode-4.0/sml-mode.el
@@ -188,7 +188,7 @@
 (defconst sml-font-lock-symbols-alist
   (append
    ;; The symbols can come from a JIS0208 font.
-   (and (fboundp 'make-char) (charsetp 'japanese-jisx0208)
+   (and (fboundp 'make-char) (fboundp 'charsetp) (charsetp 'japanese-jisx0208)
	(memq sml-font-lock-symbols '(t japanese-jisx0208))
	(list (cons "fn" (make-char 'japanese-jisx0208 38 75))
 	      (cons "andalso" (make-char 'japanese-jisx0208 34 74))
